### ISSUE TITLE
Consider renaming `components` to `blocks`

### ISSUE THREAD

**Issue by chrisvxd** [measuredco/puck#21]
In the puck config

**Comment by Danm72** [comment]
To me a Hero is a block,
Text, button, and other primitives are components.

A hero is made up of components.

So, I think it depends if you want to go down the route of block & component or just components

**Comment by Merino** [comment]
For me it's more a DX and naming thing. We're trying to make a email builder with it. 

My react project structure looks like:
- in my ./ui/component folder are button, text, inputs, 
- in my ./ui/layout folder are base templates 
- and in my ./ui/element/ folder are reusable blocks: like cards, gallery, columns, hero 

So I think if you design a layout you are more using the "blocks" than "components" group: I did not have the time to review #37

**Comment by JPrisk** [comment]
I'm interested in the semantics used in content managements and agree with @Danm72's comment above.

Lots of prior art and different names used for essentially the same thing:

- repeater with subfields ([ACF](https://www.advancedcustomfields.com/resources/repeater/))
- matrix with blocks ([Craft](https://craftcms.com/docs/4.x/matrix-fields.html))
- array of objects ([Sanity](https://www.sanity.io/docs/array-type))
- folds (a team that i'm currently working with 😅)

I looked at #37 but am not sure that I got the full context of the proposal, was the idea that "components" can have sub "components" now, and you are questioning the rename to blocks? personally I don't think this should affect the naming

In summary:
- i'm supportive of the name change to the config because `blocks` seems more accurate than `components`
- for what is shown to users in the Puck UI it'd be nice if this was customisable

**Comment by JakeSidSmith** [comment]
I personally would stick with "components", as existing design systems use the term "block" to refer to a specific type of component (e.g. in BEM/ABEM), whereas in React everything is a component regardless of depth/purpose - a block is still a component.

I think changing to block may artificially restrict what people want/will put in there. Allowing a user to compose atomic components/elements is equally as valid as allowing blocks.

I don't generally use the atom/element/etc naming scheme and just throw all my components in a components directory (where suitable), but it's not uncommon to see components split into distinct groups:

```
/components
  /atoms
  /elements
  /molecules
  /blocks
  /etc
```

References:
- https://css-tricks.com/abem-useful-adaptation-bem/
- https://medium.com/@janelle.wg/atomic-design-pattern-how-to-structure-your-react-application-2bb4d9ca5f97

**Comment by chrisvxd** [comment]
#37 has been merged for some time now, which introduces nested components. These are documented here: https://puckeditor.com/docs/integrating-puck/multi-column-layouts

I'm going to close this for now, but happy to reopen discussions if anyone feels strongly.
